### PR TITLE
tf_dataset: add unit test to verify repeat() works after cache()

### PR DESCRIPTION
**Pull request by chongxiaoc**
We can cache first iteration result then use repeat() to do multiple
epochs.

### CODE REVIEW

File: petastorm/tests/test_tf_dataset.py [hunk 1]
```diff
@@ -121,6 +121,34 @@ def test_with_dataset_repeat(synthetic_dataset, reader_factory):
                 sess.run(iterator)
 
 
+@pytest.mark.forked
+@pytest.mark.parametrize('reader_factory', ALL_READER_FLAVOR_FACTORIES)
+@create_tf_graph
+def test_with_dataset_repeat_after_cache(synthetic_dataset, reader_factory):
```
Comment (selitvin) [inline petastorm/tests/test_tf_dataset.py:127]: Not for this PR, but we should probably try using modern TF interface (here `import tensorflow.compat.v1 as tf`). That way the tests would become less bloated as we would work with eager execution mode.

File: docs/release-notes.rst [hunk 2]
```diff
@@ -6,6 +6,7 @@ Release notes
 
 Release 0.10.1 (unreleased)
 ===========================
+- `PR 674 <https://github.com/uber/petastorm/pull/674>`_: tf_dataset: add unit test to verify repeat() works after cache().
```
Comment (selitvin) [inline docs/release-notes.rst:9]: nit: usually I do not document unit-tests changes in release doc as this is not user facing (unless there is a bug fix).

**Review by selitvin — APPROVED** [review]

File: docs/release-notes.rst (hunk 2 above)
Comment (chongxiaoc) [reply docs/release-notes.rst:9]: Will remove it.

File: petastorm/tests/test_tf_dataset.py (hunk 1 above)
Comment (chongxiaoc) [reply petastorm/tests/test_tf_dataset.py:127]: I agree, it is very painful to debug without eager execution mode.

**Pull request merged**